partnerapp/src/InteractionDesignerEmbedView.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, useEffect } from "react";
+import { observer } from "mobx-react-lite";
+import { useParams, useHistory } from "react-router-dom";
+import store, { CampaignStore, Campaign } from "./campaign-store";
+
+interface InteractionDesignerEvent {
+  type: "interactiondesigner:cancel" | "interactiondesigner:save";
+  interaction_id: string;
+  organization_id: string;
+  user_id: string;
+}
+
+async function onInteractionDesignerEvent(
+  eventData: InteractionDesignerEvent,
+  store: CampaignStore,
+  campaign: Campaign,
+  history: { push: (arg0: string) => void }
+) {
+  console.log("postMessage Event handler received: ", eventData, campaign);
+  if (eventData.type === "interactiondesigner:cancel") {
+    await store.revertCampaign(campaign);
+    window.location.href = "/campaigns";
+  } else if (eventData.type === "interactiondesigner:save") {
+    // Handle interaction save
+    campaign.interactionId = eventData.interaction_id;
+    await store.publishCampaign(campaign);
+
+    setTimeout(() => {
+      // Redirect to preview view
+      window.location.href = `/campaigns/${campaign.id}/view`;
+    });
```

Comment:
Why exactly was this wrapped in a setTimeout? Can't this just be:
```suggestion
    // Redirect to preview view
    history.push(`/campaigns/${campaign.id}/view`);
```